### PR TITLE
Export pod annotations

### PR DESCRIPTION
**Pull request by ribbybibby**
I would like to construct an alert that identifies when the vault sidecar injection has failed by returning series for pods that have the injection annotation but no credentials agent container:
``` 
kube_pod_annotations{key="injector.tumblr.com/request",value=~"vault-sidecar-.*"} unless on (pod,namespace) kube_pod_container_info{container=~"vault-credentials-agent.*"}
```
This PR enables this by exporting annotations for pods.

To limit cardinality, I've added a filter for specific annotations. I've also added the same filter to namespace annotations as well.

I've tweaked the kube client creation to support loading config from more locations too.

### CODE REVIEW

**Review by hectorhuertas — APPROVED** [review]

**Review by ffilippopoulos — APPROVED** [review]
lgtm! Let's rename after merge

**Pull request merged**
